### ISSUE TITLE
Infraspecies not being assigned rank for zoological data

### ISSUE THREAD

**Issue by gdower** [Sp2000/colplus-backend#365]
For the [AG+ Scarabs ACEF export](https://www.col.plus/dataset/2050/classification), the classification is not showing infraspecies:

![Classification for Aegialia genus](https://i.gyazo.com/3cdb64b5302b3bd521afbe7ffdc830bc.png)

@mdoering, the [backend is ranking *Aegialia conferta nigrella* Brown, 1931](https://api.col.plus/dataset/2050/tree/xlL7d/children?limit=500&offset=0) as a species when it is [actually an infraspecies](http://catalogueoflife.org/col/details/species/id/977b776822ae0f292d2547506ac42df9/source/tree):

![Aegialia conferta JSON](https://i.gyazo.com/8b87dcbb6ad99fb3c3e57c54665bba34.png)

The data are correct in the AG+ ACEF export:

```
AcceptedInfraSpecificTaxa
"Scr-04-.01-.01-.00-.002-.000-.009-.b","Scr-04-.01-.01-.00-.002-.000-.009-.-","nigrella","","Brown, 1931","","accepted name","0","0","1","","","Schoolmeesters P.","2019-01-07","","",""
```

```
AcceptedSpecies
"Scr-04-.01-.01-.00-.002-.000-.009-.-","Animalia","Arthropoda","Insecta","Coleoptera","Scarabaeoidea","Aegialiidae","Aegialia","","conferta","Horn, 1871","","accepted name","0","0","1","","","Schoolmeesters P.","2019-01-07","","",""
```

My best guess is that it's because the infraspecies rank is "" because it's zoological data so CoL+ doesn't assign the subspecies rank.

**Comment by mdoering** [comment]
The problem seems to be that `Scr-04-.01-.01-.00-.002-.000-.009-.b` is the ID used for both an AcceptedSpecies and the AcceptedInfraSpecificTaxa. Because it is not unique the records get mixed up apparently. I would have expected that one of the two, probably the subspecies, gets dropped and the verbatim record flagged.

> [main] WARN  org.col.importer.neo.NeoCRUDStore - Duplicate NeoNameID Scr-04-.01-.01-.00-.002-.000-.009-.b

**Comment by gdower** [comment]
The species and infraspecies IDs look the same but they're actually different:

```
Scr-04-.01-.01-.00-.002-.000-.009-.b
Scr-04-.01-.01-.00-.002-.000-.009-.-
```

**Comment by mdoering** [comment]
but there are 2 species including the (basionym?) b suffix one:
```
"Scr-04-.01-.01-.00-.002-.000-.009-.b","Animalia","Arthropoda","Insecta","Coleoptera","Scarabaeoidea","Aegialiidae","Aegialia","","conferta","Brown, 1931","","accepted name","0","0","1","","","Schoolmeesters P.","2019-01-07","","",""
"Scr-04-.01-.01-.00-.002-.000-.009-.-","Animalia","Arthropoda","Insecta","Coleoptera","Scarabaeoidea","Aegialiidae","Aegialia","","conferta","Horn, 1871","","accepted name","0","0","1","","","Schoolmeesters P.","2019-01-07","","",""
```

You can also see that in the tree browser that the species is a duplicate with different authorship.
This looks wrong, not sure where it originated.

**Comment by mdoering** [comment]
appears to be correct in CoL: http://www.catalogueoflife.org/col/search/all/key/Aegialia+conferta/fossil/1/match/1

so maybe an ACEF export bug?

**Comment by mdoering** [comment]
You can spot that in the verbatim browser nicely:
https://www.col.plus/dataset/2050/verbatim?type=acef:AcceptedSpecies&type=acef:AcceptedInfraSpecificTaxa&acef:AcceptedTaxonID=Scr-04-.01-.01-.00-.002-.000-.009-.b&acef:AcceptedTaxonID=Scr-04-.01-.01-.00-.002-.000-.009-.b&termOp=OR

**Comment by gdower** [comment]
I bet you're right that it's an exporter bug with the AcceptedSpecies. I'll fix that tomorrow.

**Comment by gdower** [comment]
Fixed in [Sp2000/colplus-repo#27](https://github.com/Sp2000/colplus-repo/issues/27)
